server/pulp/server/db/model/__init__.py

```diff
@@ -667,6 +665,20 @@ def pre_save_signal(cls, sender, document, **kwargs):
         if not document._storage_path:
             document._storage_path = FileStorage.get_path(document)
 
+    def set_file(self, name):
+        """
+        Update the storage path to include the specified file name.
+        This should only be called when the content unit is associated with
+        a single file.
+
+        :param name: A file name.
+        :type name: str
+        """
+        if self.storage_path:
+            return
```

Comment:
Should this raise an Exception instead of returning? The method description makes it sound to me like doing this a second time would be an error.